### PR TITLE
V3/controllers target instance

### PR DESCRIPTION
**Pull request by leefrank9527**
Fixed common issues, e.g: ApplicationContext initial and getter.
Fixed issues related to TargetInstance harvest and integration with H3 Agent and Store.

### CODE REVIEW

File: src/main/java/org/webcurator/core/harvester/agent/HarvestAgentClient.java [hunk 1]
```diff
@@ -51,7 +57,7 @@ public HarvestAgentClient(String host, int port) {
     }
 
     public String baseUrl() {
-        return host + ":" + port;
+        return "http://" + host + ":" + port;
```
Comment (hannakoppelaar) [inline src/main/java/org/webcurator/core/harvester/agent/HarvestAgentClient.java]: I think there might be more hard-coded 'http' literals in the code-base, but we'll have to clean this up eventually if we want to be able to use https between components.

File: src/main/java/org/webcurator/core/harvester/coordinator/HarvestAgentListener.java [hunk 2]
```diff
@@ -33,12 +35,11 @@
     void heartbeat(HarvestAgentStatusDTO aStatus);
 
     /**
-     *
-     * @param haHost
-     * @param haPort
-     * @param haService
+     * Recover connection
```
Comment (hannakoppelaar) [inline src/main/java/org/webcurator/core/harvester/coordinator/HarvestAgentListener.java]: This is a potentially confusing comment: this method recovers crawl jobs

**Comment by obrienben** [comment]
@hannakoppelaar I've tested now against the PRs for store and harvest-agent-h3, and they are building and passing tests (I pushed one commit for the harvest agent to fix a broken test).
I'll take a look at the failing tests in Core

File: src/main/java/org/webcurator/core/harvester/agent/HarvestAgentClient.java (hunk 1 above)
Comment (leefrank9527) [reply src/main/java/org/webcurator/core/harvester/agent/HarvestAgentClient.java]: Accept. I would create an abstract client class and define the protocol scheme, host and port of a client.

**Comment by leefrank9527** [comment]
@hannakoppelaar There would be some compilation fails for harvester-h1 because it hasn't been integrated with the new core component. For harvester-h3, I had commited all related changes, only some local configuration files excluded.

File: src/main/java/org/webcurator/core/harvester/coordinator/HarvestAgentListener.java (hunk 2 above)
Comment (leefrank9527) [reply src/main/java/org/webcurator/core/harvester/coordinator/HarvestAgentListener.java]: Yes. I tried to refine as this:
 * The <code>HarvestAgentListener</code> is a service that 
 * listens for messages from a <code>Crawler</code> and
 * forward the messages to a <code>Store</code> and a
 * <code>Webapp</code>.

**Comment by obrienben** [comment]
@leefrank9527 are all the unit tests now passing for you in Core? I'm still seeing test failures in HarvestAgentManagerImplTest.

**Pull request merged**